src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java

```diff
@@ -143,9 +145,17 @@ private boolean shouldSab() {
     }
 
     private boolean shouldRsb() {
-        if (!main.config.LOOT.RSB.ENABLED || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
-        if (usedRsb < System.currentTimeMillis() - main.config.LOOT.RSB.AMMO_REFRESH) usedRsb = System.currentTimeMillis();
-
+        if (!main.config.LOOT.RSB.ENABLED || main.config.LOOT.RSB.KEY == null
+                || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
+        String itemId = main.facadeManager.slotBars.standardBar.slots.stream()
+                .filter(slot -> Objects.equals(main.facadeManager.settings.getCharCode(valueOf
+                        ("SLOTBAR_" + (slot.slotNumber == 10 ? 0 : slot.slotNumber))), main.config.LOOT.RSB.KEY))
+                .findFirst().map(slot -> slot.item != null ? slot.item.id : null).orElse(null);
```

Comment:
it was warning me when i try to do `.map(slot -> slot.item.id)` seperatng it to two maps solves that issue, or it was never an issue?

Reply:
Because you're using an `Optional`, doing `.map(a -> null)` will not be any exception, it will simply become an `empty` optional and when you later do another `.map` it won't throw an exception, it will simply continue to be an `empty` optional.

however if you try to `.map(a -> null.something)` it will throw NPE, as you tried to get some inner variable of a null reference